mlflow/xgboost.py

```diff
@@ -286,6 +287,30 @@ def autolog(importance_types=["weight"]):  # pylint: disable=W0102
     import xgboost
     import numpy as np
 
+    class _InputExampleInfo:
+        def __init__(self, input_example=None, error_msg=None):
+            self.input_example = input_example
+            self.error_msg = error_msg
+
+    def __init__(self, *args, **kwargs):
+        data = args[0] if len(args) > 0 else kwargs.get("data")
+
+        if data is not None:
+            original = gorilla.get_original_attribute(xgboost.DMatrix, "__init__")
+
+            input_example_info = None
+            try:
+                if isinstance(data, str):
+                    raise Exception("The input data was of type string.")
+
+                input_example_info = _InputExampleInfo(input_example=deepcopy(data[:5]))
```

Comment:
can we pull `5` into a variable defined in `autologging_utils.py` and use that variable throughout the various autologging integrations that support signature / input example collection? (i.e. sklearn, lightgbm, xgboost). cc @harupy